src/init.cpp

```diff
@@ -611,6 +609,12 @@ static void BlockNotifyCallback(bool initialSync, const CBlockIndex *pBlockIndex
 }
 #endif
 
+static void StakingPoolCallback(bool initialSync, const CBlockIndex *pBlockIndex) {
+    if (initialSync || !pBlockIndex)
+        return;
+    CStakingPool::getInstance()->increaseBalanceForNewBlock(pBlockIndex->nHeight);
+}
```

Comment:
That's not a good place for this. I think the best place to increment staking pool balance would be at `CChainState::ConnectBlock`. Moreover, CChainstate would be the best handler for staking pool as a whole

Reply:
I moved increase and decrease staking balance feature into `ConnectBlock` and `DisconnectBlock` methods. I added also functional test in `feature_staking_pool.py` to check correctness of the implementation.